### PR TITLE
Bpk 1247 add shadow tokens

### PR DESCRIPTION
**Pull request by k0nserv**
Here's what it looks like

simulator screen shot - iphone 8 - 2018-02-08 at 16 10 23

### CODE REVIEW

File: Backpack/Classes/Shadow/Generated/BPKShadow.m [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    layer.shadowRadius = self.radius;
+    layer.shadowOpacity = self.opacity;
+}
+
+
++ (instancetype)shadowSm {
+    static dispatch_once_t shadowSmOnceToken;
+    static BPKShadow *shadowSm;
+
+    dispatch_once(&shadowSmOnceToken, ^{
```
Comment (k0nserv) [inline Backpack/Classes/Shadow/Generated/BPKShadow.m]: This pattern is a thread safe way to ensure only a single instance of a value gets created.

File: Example/Tests/BPKShadowTest.m [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        XCTAssert([BPKShadow respondsToSelector:selector]);
+        XCTAssertNotNil([BPKShadow performSelector:selector]);
+    }
+}
+
+- (void)testStylesAreCached {
+    BPKShadow *shadow1 = [BPKShadow shadowSm];
+    BPKShadow *shadow2 = [BPKShadow shadowSm];
+
+    XCTAssertTrue(shadow1 == shadow2, @"BPKShadow should cache shadow styles");
```
Comment (k0nserv) [inline Example/Tests/BPKShadowTest.m]: This verifies we never create more than a single value per shadow type by doing pointer comparison. Object comparison in objective-c is done with `isEqual:`

File: README.md [hunk 3]
```diff
@@ -65,6 +65,20 @@ CGFloat myAppBorderRadiusSmall = BPKBorderRadiusSm;
...
+### Backpack/Shadow
+
+`Backpack/Shadow` contains the Backpack shadow definitions in the class `BPKShadow`. Add `Backpack/Shadow` to your `Podfile` or `podspec` and then use it as follows.
+
+```objective-c
+#import <Backpack/Shadow.h>
+
+BPKShadow *shadow = [BPKShadow shadowSm];
+// BPKShadow *shadowLg = [BPKShadow shadowLg];
+// BPKShadow *shadowXl = [BPKShadow shadowXl];
```
Comment (shaundon) [inline README.md]: Just want to confirm that these are supposed to be commented out?

Comment (k0nserv) [reply README.md]: Yeah

**Pull request merged**